### PR TITLE
Ag cloudformation riffraff

### PR DESCRIPTION
**Pull request by fredex42**
cloudformation and riffraff setup

### CODE REVIEW

File: conf/deploy.annotated-json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (akash1810) [inline conf/deploy.annotated-json]: don't think we'd need this in the repo?

File: test/ApplicationSpec.scala [hunk 2]
```diff
@@ -19,12 +19,12 @@ class ApplicationSpec extends Specification {
...
     }
 
-    "render the index page" in new WithApplication{
-      val home = route(FakeRequest(GET, "/")).get
-
-      status(home) must equalTo(OK)
-      contentType(home) must beSome.which(_ == "text/html")
-      contentAsString(home) must contain ("Your new application is ready.")
-    }
+//    "render the index page" in new WithApplication{
```
Comment (akash1810) [inline test/ApplicationSpec.scala]: should this be commented out? if yes, then just delete it entirely.

File: build.sbt [hunk 3]
```diff
@@ -1,17 +1,58 @@
...
 
-lazy val root = (project in file(".")).enablePlugins(PlayScala)
+
+unmanagedResourceDirectories in Test <+=  baseDirectory ( _ /"target/web/public/test" )
+
+riffRaffPackageType := (packageZipTarball in Universal).value
+
+def env(key: String): Option[String] = Option(System.getenv(key))
+
+/* //commenting these out as they should be set above
```
Comment (akash1810) [inline build.sbt]: lets remove these lines if not being used

File: build.sbt [hunk 4]
```diff
@@ -1,17 +1,58 @@
+import sbt.Keys._
+
 name := "live-stream-creator"
 
 version := "1.0"
 
+/*i am not sure if we need this or not?*/
+//lazy val root = (project in file(".")).enablePlugins(PlayScala)
```
Comment (akash1810) [inline build.sbt]: `lazy val root` is just a little bit nicer as there are no backticks. Lets choose one or the other and delete the unused one.

File: cfn/cfn.json [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+                    "subnet-d58a938c",
+                    "subnet-577ca433",
+                    "subnet-a0d211d6"
+                ],
+                "MaxSize": "4",
+                "MinSize": "1",
+                "Tags": [
+                    {
+                        "Key": "Stage",
+                        "Value": "CODE",
```
Comment (akash1810) [inline cfn/cfn.json]: can this be a param, so we can reuse the template for PROD too?

File: cfn/cfn.json [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+                        "PropagateAtLaunch": "true"
+                    },
+                    {
+                        "Key": "Stack",
+                        "Value": "multimedia",
+                        "PropagateAtLaunch": "true"
+                    },
+                    {
+                        "Key": "Role",
```
Comment (akash1810) [inline cfn/cfn.json]: what's this tag for?

File: cfn/cfn.json [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+        },
+        "ASLC13W7C": {
+            "Type": "AWS::AutoScaling::LaunchConfiguration",
+            "Properties": {
+                "IamInstanceProfile": {
+                    "Ref": "IAMIP38VNO"
+                },
+                "ImageId": "ami-c79bebb4",
+                "InstanceType": "t2.micro",
+                "KeyName": "AndyTestKey",
```
Comment (akash1810) [inline cfn/cfn.json]: lets use a 'real' key

File: cfn/cfn.json [hunk 8]
```diff
@@ -0,0 +1,375 @@
...
+                "UserData": {
+                    "Fn::Base64": {
+                        "Fn::Join": [
+                            "",
+                            [
+                                "#!/bin/bash\n",
+                                "/opt/features/ssh-keys/initialise-keys-and-cron-job.sh -b github-team-keys -t multimedia -l || true\n",
```
Comment (akash1810) [inline cfn/cfn.json:188]: awesome!

File: cfn/cfn.json [hunk 9]
```diff
@@ -0,0 +1,375 @@
...
+                        "PolicyName": "GetTeamKeysPolicy",
+                        "PolicyDocument": {
+                            "Statement": [
+                                {
+                                    "Effect": "Allow",
+                                    "Action": [
+                                        "s3:GetObject"
+                                    ],
+                                    "Resource": [
+                                        "arn:aws:s3:::github-team-keys/*"
```
Comment (akash1810) [inline cfn/cfn.json:299]: should this bucket be a param?

File: cfn/cfn.json [hunk 10]
```diff
@@ -0,0 +1,375 @@
...
+                }
+            }
+        },
+        "EC2SG2PWPP": {
+            "Type": "AWS::EC2::SecurityGroup",
+            "Properties": {
+                "GroupDescription": "HTTP access from VPC",
+                "SecurityGroupIngress": [
+                    {
+                        "CidrIp": "10.255.10.0/24",
```
Comment (akash1810) [inline cfn/cfn.json:343]: who is this?

File: credentials.cipher [hunk 11]
```diff
@@ -0,0 +1,4 @@
+Salted__aOS�_�Pu"��&!�(>�x��)���<ɕ
```
Comment (akash1810) [inline credentials.cipher]: this makes me a little nervous in a public repo, do we need it?

File: cfn/cfn.json (hunk 5 above)
Comment (fredex42) [reply cfn/cfn.json]: The "Stage" tag, you mean?

File: cfn/cfn.json (hunk 6 above)
Comment (fredex42) [reply cfn/cfn.json]: Stack? I think it was needed for riffraff?

File: cfn/cfn.json (hunk 7 above)
Comment (fredex42) [reply cfn/cfn.json]: :) sure. Parameterise?

File: cfn/cfn.json (hunk 9 above)
Comment (fredex42) [reply cfn/cfn.json:299]: hmmm. guess we should ask @philmcmahon? i assumed that it was a static one

File: cfn/cfn.json (hunk 10 above)
Comment (fredex42) [reply cfn/cfn.json:343]: it's the VPC within which the instance is launched. only the LB (which lives in the same VPC) needs access; so it feels nicer to not grant extra access

File: credentials.cipher (hunk 11 above)
Comment (fredex42) [reply credentials.cipher]: well, it was recommended as "the" way to integrate circleci here - https://github.com/circleci/encrypted-files (linked from https://circleci.com/docs/environment-variables/#keeping-encrypted-environment-variables-in-source-code) - but we could put them directly into circle i guess?

**Comment by akash1810** [comment]
👍

**Pull request merged**